lib/redmine/menu_manager/top_menu_helper.rb

```diff
@@ -215,6 +215,8 @@ def more_top_menu_items
 
   def module_top_menu_item_groups
     items = more_top_menu_items
+    return items if items.empty?
```

Comment:
Even though an empty array is an empty array in ruby, I would suggest not returning the empty `items` array here, but the `item_groups` array. You can do this in a way,  that add `&` in line 223: `unless items.first&.heading?`. If you do it that way the `reduce` is operating again on an empty list or containing a single item.

Let me know if I should explain more in depth.

Reply:
Doing that gives us empty menu dropdown as `unless items.first&.heading?` adds to item_groups `[{:title=>nil, :items=>[]}]`

![Screenshot_20241204_195215](https://github.com/user-attachments/assets/e109e14e-6448-432a-a9b8-768f196588a8)

Reply:
As a variant?
```ruby
def module_top_menu_item_groups
    item_groups = []
    items = more_top_menu_items
    return item_groups if items.blank?
```